src/metabase/email/messages.clj

```diff
@@ -58,39 +58,45 @@
 ;; HACK: temporary workaround to postal requiring a file as the attachment
 (defn- write-byte-array-to-temp-file
   [img-bytes]
-  (let [file (java.io.File/createTempFile "metabase_pulse_image_" ".png")
-        fos (new java.io.FileOutputStream file)]
-    (.deleteOnExit file)
-    (.write fos img-bytes)
-    (.close fos)
+  (let [file (doto (java.io.File/createTempFile "metabase_pulse_image_" ".png")
+               .deleteOnExit)]
+    (with-open [fos (java.io.FileOutputStream. file)]
+      (.write fos (bytes img-bytes)))
     file))
 
 (defn- find-byte-array
   [needle haystack]
-  (first (keep-indexed #(when (java.util.Arrays/equals %2 needle) %1) haystack)))
+  (first (keep (fn [item]
+                 (when (java.util.Arrays/equals (bytes item) (bytes needle))
+                   true))
+               haystack)))
+
+(defn- render-image [images-atom bytes]
+  (let [index (or (find-byte-array bytes @images-atom)
+                  (count @images-atom))]
+    (when (= index (count @images-atom))
+      (swap! images-atom conj bytes))
+    (str "cid:IMAGE" index)))
 
 (defn render-pulse-email
   "Take a pulse object and list of results, returns an array of attachment objects for an email"
   [pulse results]
   (let [images       (atom [])
-        render-img   (fn [bytes]
-                        (let [index (or (find-byte-array bytes @images) (count @images))]
-                          (if (= index (count @images)) (reset! images (conj @images bytes)))
```

Comment:
`swap!` lets you atomicly apply a function to an atom, use that instead of `reset!` whenever possible